hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/logical/CalcEliminateDropLateItemsRule.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.opt.logical;
+
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelRule;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.Calc;
+import org.apache.calcite.rel.rules.TransformationRule;
+import org.apache.calcite.rex.RexInputRef;
+import org.apache.calcite.rex.RexNode;
+import org.immutables.value.Value;
+
+import java.util.List;
+
+import static com.hazelcast.jet.sql.impl.opt.Conventions.LOGICAL;
+
+/**
+ * Logical rule that eliminates {@link DropLateItemsLogicalRel}
+ * under {@link Calc}, if calc doesn't project watermarked field.
+ * <p>
+ * Before:
+ * <pre>
+ * Calc[...]
+ *   DropLateItemsRel[...]
+ * </pre>
+ * After:
+ * <pre>
+ * Calc[...]
+ * </pre>
+ * <p>
+ */
+@Value.Enclosing
+public class CalcEliminateDropLateItemsRule extends RelRule<RelRule.Config> implements TransformationRule {
```

Comment:
This rule is wrong. The `drop-rel` is a kind of a filter. We cannot simply remove the filter just because something after it eliminates the column by which it filtered.